src/Sulu/Bundle/AdminBundle/Resources/js/components/Select/Select.js

```diff
@@ -118,6 +180,73 @@ class Select<T> extends React.Component<Props<T>> {
 
     handleOptionListClose = this.closeOptionList;
 
+    @computed get highestButtonIndex() {
+        const max = Math.max(...Array.from(this.buttonRefs.keys()));
+        return max;
+    }
+
+    @action requestFocus = (focusedElementIndex: number) => {
+        if (focusedElementIndex < 0 || focusedElementIndex > this.highestButtonIndex) {
+            return;
+        }
+
+        if (this.buttonRefs.has(focusedElementIndex)) {
+            this.focusedElementIndex = focusedElementIndex;
+            const ref = this.buttonRefs.get(focusedElementIndex);
+
+            if (ref) {
+                ref.focus();
+            }
+        }
+    };
+
+    handleRequestFocus = (focusedElementIndex: number) => () => {
+        this.requestFocus(focusedElementIndex);
+    };
+
+    @action handleKeyDown = (event: KeyboardEvent) => {
+        if (!this.open) {
+            return;
+        }
+
+        event.preventDefault();
+
+        if (event.key === 'Escape') {
+            this.closeOptionList();
+
+            return;
+        }
+
+        let focusedElementIndex = this.focusedElementIndex;
+
+        if (event.key === 'ArrowUp') {
+            do {
+                --focusedElementIndex;
+            } while (!this.buttonRefs.has(focusedElementIndex) && focusedElementIndex > 0);
```

Comment:
Is that some kind of busy wait? Or why do we need a loop here? Is maybe that the reason for some of the performance issues you are experiencing?

Reply:
think this is needed to skip `<Divider>` children, right? But it might be easier to understand when using something like `focusedElementIndex  = Math.min(...availableIndices.filter(i => i > this.focusedElementIndex), 0)`

Reply:
yes, exactly

Reply:
will try your suggestions @nnatter

Reply:
fixed